src/aggregation/metric/extended_stats.rs

```diff
@@ -0,0 +1,1177 @@
+    pub fn merge_fruits(&mut self, other: IntermediateExtendedStats) {
+        if other.intermediate_stats.count == 0 {
+            return;
+        }
+        if self.intermediate_stats.count == 0 {
+            let _ = mem::replace(self, other);
+            return;
+        }
+        let new_count = self.intermediate_stats.count + other.intermediate_stats.count;
+        let delta = other.mean - self.mean;
+        self.sum_of_squares += other.sum_of_squares
+            + delta
+                * delta
+                * self.intermediate_stats.count as f64
+                * other.intermediate_stats.count as f64
+                / new_count as f64;
+        self.mean = (self.intermediate_stats.sum as f64 + other.intermediate_stats.sum as f64)
+            / new_count as f64;
+        self.sum_of_squares_elastic += other.sum_of_squares_elastic;
+        self.delta_sum_for_squares_elastic += other.delta_sum_for_squares_elastic;
+        self.intermediate_stats
+            .merge_fruits(other.intermediate_stats);
+    }
+
+    /// Computes the final stats value.
+    pub fn finalize(&self) -> Box<ExtendedStats> {
+        let (min, max, avg, sum_of_squares) = if self.intermediate_stats.count == 0 {
+            (None, None, None, None)
+        } else {
+            (
+                Some(self.intermediate_stats.min),
+                Some(self.intermediate_stats.max),
+                Some(self.mean),
+                Some(self.sum_of_squares_elastic),
+            )
+        };
+        let (variance, variance_sampling) = if self.intermediate_stats.count <= 1 {
+            (None, None)
+        } else {
+            (
+                Some(self.sum_of_squares / self.intermediate_stats.count as f64),
+                Some(self.sum_of_squares / (self.intermediate_stats.count - 1) as f64),
+            )
+        };
+        let std_deviation = variance.map(|v| v.sqrt());
+        let std_deviation_sampling = variance_sampling.map(|v| v.sqrt());
+        let std_deviation_bounds = if std_deviation.is_none() {
+            None
+        } else {
+            let upper = self.mean + std_deviation.unwrap() * self.sigma;
+            let lower = self.mean - std_deviation.unwrap() * self.sigma;
+            let upper_sampling = self.mean + std_deviation_sampling.unwrap() * self.sigma;
+            let lower_sampling = self.mean - std_deviation_sampling.unwrap() * self.sigma;
+            Some(StandardDeviationBounds {
+                upper,
+                lower,
+                upper_sampling,
+                lower_sampling,
+                upper_population: upper,
+                lower_population: lower,
+            })
+        };
+        Box::new(ExtendedStats {
+            count: self.intermediate_stats.count,
+            sum: self.intermediate_stats.sum,
+            min,
+            max,
+            avg,
+            sum_of_squares,
+            variance,
+            variance_population: variance,
```

Comment:
why are there two fields with the same value?

Reply:
Basically because of ElasticSearch implementation, the documentation (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-extendedstats-aggregation.html) states: `The std_deviation and variance are calculated as population metrics so they are always the same as std_deviation_population and variance_population respectively.`